programs/uxd/src/instructions/compute_depositories_targets.rs

```diff
@@ -0,0 +1,261 @@
+        mercurial_vault_depository_raw_target,
+        mercurial_vault_depository_cap,
+    )?;
+    let credix_lp_depository_availability = ctx
+        .accounts
+        .compute_availability(credix_lp_depository_raw_target, credix_lp_depository_cap)?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 3
+    // -- Compute the combined overflow of all depositories
+    // -- Compute the combined availability of all depositories
+    // ---------------------------------------------------------------------
+
+    // Compute total amount that doesn't fit within depositories cap
+    let total_overflow = ctx.accounts.compute_total(
+        mercurial_vault_depository_overflow,
+        credix_lp_depository_overflow,
+    )?;
+    // Compute total amount that doesn't fit within depositories cap
+    let total_availability = ctx.accounts.compute_total(
+        mercurial_vault_depository_availability,
+        credix_lp_depository_availability,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Compute the final target based off of the logic:
+    // -- * FinalTarget = raw_target - overflow + ExtraBonus
+    // -- * ExtraBonus = total_overflow * (availability / total_availability)
+    // -- In other words:
+    // -- * The final target is capped at the cap
+    // -- * Any amount overflowing that is transfered to others depositories
+    // -- * Depositories with available space will receive overflows
+    // ---------------------------------------------------------------------
+
+    // Compute the final targets for each depository
+    let mercurial_vault_depository_final_target = ctx.accounts.compute_final_target(
+        mercurial_vault_depository_raw_target,
+        mercurial_vault_depository_overflow,
+        mercurial_vault_depository_availability,
+        total_overflow,
+        total_availability,
+    )?;
+    let credix_lp_depository_final_target = ctx.accounts.compute_final_target(
+        credix_lp_depository_raw_target,
+        credix_lp_depository_overflow,
+        credix_lp_depository_availability,
+        total_overflow,
+        total_availability,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 5
+    // -- Update the on-chain depositories with computed results
+    // ---------------------------------------------------------------------
+
+    // Update onchain accounts
+    mercurial_vault_depository.redeemable_amount_under_management_target_amount =
+        mercurial_vault_depository_final_target;
+    credix_lp_depository.redeemable_amount_under_management_target_amount =
+        credix_lp_depository_final_target;
+
+    // Success
+    Ok(())
+}
+
+// Into functions
+impl<'info> ComputeDepositoriesTargets<'info> {
+    // Compute a simple raw target: raw_target = total_circulating_supply * weight
```

Comment:
nit: comment could be deleted

- Repeating function name in comment have no value.
- Function content is self-explanatory for the formula.